src/withVueRouter.ts

```diff
@@ -71,41 +75,43 @@ export const withVueRouter = (
   name: 'withVueRouter',
   parameterName: 'withVueRouter',
 
-  wrapper: (storyFn: LegacyStoryFn, context: StoryContext) => {
-    /* setup router var */
-    let router
+  wrapper: (storyFn: StoryFn, context: StoryContext) => {
+    setup(app => {
```

Comment:
Hi @NickMcBurney, 

I think I found the issue in your code.
On line 113 you return the storyFn.

This return needs to be outside of the `setup` fn.
For what it's worth, I actually took your code and applied it to the storybookjs/addon-kit storybook7 branch.
https://github.com/storybookjs/addon-kit/tree/sb-7

I also decided to use vite and plan to refactor the vue code to script setup